src/app/static/src/tests/components/forms/dynamicFormChecker.test.ts

```diff
@@ -0,0 +1,309 @@
+            controlSections: [{
+                label: "l1", controlGroups: [{}]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise missing control name as invalid", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{required: false, type: "number"}]
+                }]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise invalid helpText", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{name: "i1", helpText: false, required: false, type: "number"}]
+                }]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise valid helpText", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{name: "i1", helpText: "Some help text", required: false, type: "number"}]
+                }]
+            }]
+        })).toBe(true);
+    });
+
+    it("should recognise wrong control type as invalid", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{name: "i1", required: false, type: "nonsense"}]
+                }]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise missing required flag as invalid", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{name: "i1", type: "number"}]
+                }]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise required property of wrong type as invalid", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1", controlGroups: [{
+                    controls: [{name: "i1", required: "yes", type: "number"}]
+                }]
+            }]
+        })).toBe(false);
+    });
+
+    it("should recognise valid number input", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1",
+                controlGroups: [{
+                    controls: [{
+                        name: "i1",
+                        required: true,
+                        type: "number"
+                    }]
+                }]
+            }]
+        })).toBe(true);
+    });
+
+    it("should recognise valid number input with value", () => {
+        expect(isDynamicFormMeta({
+            controlSections: [{
+                label: "l1",
+                controlGroups: [{
+                    controls: [{
+                        name: "i1",
+                        required: true,
+                        type: "number",
+                        value: 10
```

Comment:
Maybe add test that "10" is recognised as invalid?

Reply:
Done. Although maybe it would be better to accept string values that can be parsed as numbers :thinking:

Reply:
In fact the html input will accept "10", so we should as well. Updated accordingly: https://github.com/mrc-ide/hint/pull/138/commits/76a54a7bd35ea286a5c90da113522fe3087e366c